includes/Core/Modules/REST_Modules_Controller.php

```diff
@@ -480,6 +480,42 @@ private function get_rest_routes() {
 					),
 				)
 			),
+			new REST_Route(
+				'modules/(?P<slug>[a-z0-9\-]+)/data/data-available',
+				array(
+					array(
+						'methods'             => WP_REST_Server::CREATABLE,
+						'callback'            => function( WP_REST_Request $request ) {
+							$slug = $request['slug'];
+							try {
+								$module = $this->modules->get_module( $slug );
+							} catch ( Exception $e ) {
+								return new WP_Error( 'invalid_module_slug', __( 'Invalid module slug.', 'google-site-kit' ), array( 'status' => 404 ) );
+							}
+
+							if ( ! $this->modules->is_module_connected( $slug ) ) {
+								return new WP_Error( 'module_not_connected', __( 'Module is not connected.', 'google-site-kit' ), array( 'status' => 400 ) );
```

Comment:
The module not being connected is an internal reason for the request to fail, so a 4xx error wouldn't be appropriate here. It doesn't make a big difference for our use, but let's use a 500 instead 👍